inetum-realjobs-backend/src/main/java/world/inetum/realdolmen/realjobs/services/VacancyService.java

```diff
@@ -42,4 +46,23 @@ public Vacancy addVacancy(Vacancy vacancy) {
                 .and(withFunctionTitle(functionTitle))
         );
     }
+
+    public boolean checkVacancyAccess(Vacancy vacancy, final String email) {
+        boolean hasAccess = false;
+
+        if (vacancy.getRecruiter() != null) {
+            hasAccess = vacancy.getRecruiter().getEmail().equals(email);
+        } else if (vacancy.getCompany() != null) {
+            Optional<Account> accountOptional = accountService.findByEmail(email);
+
+            if (accountOptional.isPresent()
+                    && accountOptional.get() instanceof Recruiter recruiter
+                    && recruiter.getCompany() != null) {
+                hasAccess = vacancy.getCompany().getId().equals(recruiter.getCompany().getId());
+            }
+        }
```

Comment:
So if the vacancy has a linked recruiter, the others in the company can't access the vacancy?

Reply:
How I understood how vacancies should work there are 2 possibilities:

- a vacancy is linked to a recruiter
- a vacancy is linked to a company

Either recruiter is null, or company is null.